docs/guides/flax_fundamentals/rng_guide.md

```diff
@@ -0,0 +1,797 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+---
+
+# Randomness and PRNGs in Flax
+
++++
+
+In this guide, you will learn how Flax uses [JAX's explicit pseudorandom number generator (PRNG) keys](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#jax-prng) to emulate randomness, and adds some additional features to make it easier for users to thread PRNG keys through different Flax `Module`s.
+
+If you are new to JAX PRNG keys or need a refresher, check out:
+- [JAX 101: PRNGs in JAX](https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html)
+- [🔪 JAX - The Sharp Bits 🔪: Random Numbers](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#random-numbers).
+
++++
+
+## Setup
+
++++
+
+Install or upgrade Flax, and then import some necessary dependencies.
+
+**Note:** This guide uses the `--xla_force_host_platform_device_count=8` flag to emulate multiple devices in a CPU environment in a Google Colab/Jupyter Notebook. You don’t need this if you are already using a multi-device Google Cloud TPU environment, for example, on Google Cloud or in a Kaggle VM with a TPU.
+
+```{code-cell}
+:outputId: bb13d0ba-f908-4746-e4d3-f5916e0670ff
+:tags: [skip-execution]
+
+!pip install -q flax
+```
+
+```{code-cell}
+import os
+os.environ["XLA_FLAGS"] = '--xla_force_host_platform_device_count=8'
+```
+
+```{code-cell}
+import flax, flax.linen as nn
+import jax, jax.numpy as jnp
+from jax.sharding import Mesh, PartitionSpec, NamedSharding
+from jax.experimental import mesh_utils
+from jax.experimental.shard_map import shard_map
+
+import hashlib
+```
+
+```{code-cell}
+:outputId: ab5e0d3b-2d51-44ee-b823-d152ad1a10b2
+
+jax.devices()
+```
+
+Set Jax config variable `jax_threefry_partitionable` to `True`. This will be the default value in the future and makes the RNG more efficiently auto-parallelizable under `jax.jit`. See this [JAX discussion](https://github.com/google/jax/discussions/18480) for more details.
```

Comment:
Nit:
```suggestion
Set the JAX config variable `jax_threefry_partitionable` to `True`. This will be the default value in the future and makes the PRNG more efficiently auto-parallelizable under `jax.jit`. Refre to this [JAX discussion](https://github.com/google/jax/discussions/18480) for more details.
```